### ISSUE TITLE
[libraries] Allow to stamping block

### ISSUE THREAD

**Issue by dekrain** [jmoenig/Snap--Build-Your-Own-Blocks#1494]
I have idea to add a library/native block that stamping block on screen. It will useful for some presentations/visual languages created by users.
It can be in `tools.xml`. `<pen>: stamp block %shape %color %spec on the screen`
Spec can be e.g. `say [text] for (3) secs`.

**Comment by brianharvey** [comment]
We're trying to keep tools.xml as small as possible, but I'd definitely be happy to see this in a separate library.  I'd use it often myself!

Make sure your spec format allows for Procedure-type input slots (I suggest (()), [[][]], and <<>> as the notation), list slots (suggest {}), and C-slots (I dunno, there's really no good ASCII representation; it should be ⊂ (subset) or something).

**Comment by dekrain** [comment]
Ok, I start working on this. I'll notice about my work when I'll make this.

**Comment by cycomachead** [comment]
Why have it take in a spec, why not just take in a ringified block?

**Comment by dekrain** [comment]
Ouch, why I didn't think about it?

**Comment by dekrain** [comment]
But it can't make custom block spec

**Comment by cycomachead** [comment]
Well you could always build the blocks yourself, unless you want to programmatically create a block. :)

Well I guess it's time to build the block building block. :D

-- 
Michael Ball
From my iPhone
michaelball.co

**Comment by brianharvey** [comment]
Right, as I think about it, there are two different uses for this idea.  @dekrain wants interactive documentation, putting blocks on the stage and animating them; but the DEFINE block that takes a spec and a thunk, and makes a block out of them, will use the same translation.  So you might as well do both in the same library.

> unless you want to programmatically create a block. :)

I _totally_ want to do that!  All the time!
